conda-store-server/conda_store_server/_internal/worker/build.py

```diff
@@ -335,18 +335,15 @@ def build_conda_environment(db: Session, conda_store, build):
 
 
 def solve_conda_environment(db: Session, conda_store, solve: orm.Solve):
-    settings = conda_store.get_settings(db=db)
-
     solve.started_on = datetime.datetime.utcnow()
     db.commit()
 
-    context = action.action_solve_lockfile(
-        conda_command=settings.conda_command,
-        specification=schema.CondaSpecification.parse_obj(solve.specification.spec),
+    _, locker = conda_store.lock_plugin()
```

Comment:
Same here about the per-session configurability fo the locker engine.

Reply:
Yes, this can be supported! In `CondaStore:lock_plugin` we are currently pulling the configured plugin from the traitlets config. 

```
    def lock_plugin(self) -> tuple[str, lock.LockPlugin]:
        """Returns the configured lock plugin"""
        lock_plugin = self.plugin_manager.lock_plugin(name=self.lock_plugin_name)
        locker = lock_plugin.backend()
        return lock_plugin.name, locker
```

This should be extended to include a query to the settings table to get the right plugin for the particular run. For example something like:

```
    def lock_plugin(self, namespace: str, environment_name: str) -> tuple[str, lock.LockPlugin]:
        """Returns the configured lock plugin"""
        settings = self.get_settings(db, namespace, environment_name)
        lock_plugin = self.plugin_manager.lock_plugin(name=settings.lock_plugin_name)
        locker = lock_plugin.backend()
        return lock_plugin.name, locker
```


dependant on https://github.com/conda-incubator/conda-store/issues/1005

Reply:
Good, let's leave a `# TODO` comment here then!